siphon/aws_goes.py

```diff
@@ -0,0 +1,185 @@
+            the comparison of obj.key <= end_key across hour or day boundaries
+            will fail, especially for ABI products, where there is a channel to
+            select.
+        """
+        path, prod_mode, nc_basename = product.key_components()
+        print("prod_mode is", prod_mode)
+        end_key = product.with_start_time(end)
+
+        # Get a list of files that have the proper prefix up to the hour
+        return list(itertools.takewhile(lambda obj: (obj.key <= end_key),
+                                        self._get_iter(start, product)))
+
+    def get_range(self, start, end, product):
+        in_range = []
+        for t0, t1 in pairwise(gen_day_chunks(start, end)):
+            this_range = self.get_range_in_hour_chunks(t0, t1, product)
+            in_range.extend(this_range)
+        return in_range
+
+
+class GOESProduct(object):
+    def __init__(self, **kwargs):
+        self.sector = 'conus'
+        self.satellite = 'goes16'
+        self.typ = 'ABI'
+        self.channel = 1
+        self.mode = 3
+        self.keypath = '{prod_id}'
+        self.__dict__.update(kwargs)
+
+    def key_components(self):
+        env = 'OR'
+        sat = {'goes16': 'G16', 'goes17': 'G17'}[self.satellite]
+
+        if self.typ == 'ABI':
+            sectors = {'conus': 'C', 'meso1': 'M1', 'meso2': 'M2', 'full': 'F'}
+            sector = sectors[self.sector]
+            prod_id = 'ABI-L1b-Rad{sector}'.format(sector=sector)
+            prod_code = '-M{mode}C{channel:02d}'.format(sector=sector,
+                                                        mode=self.mode,
+                                                        channel=self.channel)
+            prod_mode = prod_id + prod_code
+        elif self.typ == 'GLM':
+            prod_id = 'GLM-L2-LCFA'
+            prod_mode = prod_id
+        else:
+            raise ValueError('Unhandled data type: {}'.format(self.typ))
+        path = prod_id
+        nc_basename = '{env}_{prodid}_{sat}'.format(env=env, prodid=prod_mode,
+                                                    sat=sat)
+        return path, prod_mode, nc_basename
+
+    def prefix(self, time):
+        path, prod_mode, nc_basename = self.key_components()
+        return path + '/{time:%Y/%j/%H}/'.format(time=time) + nc_basename
+
+    def __str__(self):
+        path, prod_mode, nc_basename = self.key_components()
+        return path
+
+    __repr__ = __str__
+
+    def with_start_time(self, time):
+        path, prod_mode, nc_basename = self.key_components()
+        base = self.prefix(time)
+        return (base + '_s{time:%Y%j%H%M%S}').format(time=time)
+
+
+def netcdf_from_s3(s3obj):
+    """ Download the data and open (in memory) with netCDF """
+    with NamedTemporaryFile(suffix='.nc') as temp:
+        # Create a temporary netCDF file to work around bug in netCDF C 4.4.1.1
+        # We shouldn't actually need any file on disk.
```

Comment:
Is there an existing bug report for this? Is this fixed now?

Reply:
I think this is fixed now.